src/sentry/models/file.py

```diff
@@ -109,3 +120,47 @@ def getfile(self):
 
         storage = self.get_storage()
         return storage.open(self.path)
+
+
+class File(Model):
+    __core__ = False
+
+    name = models.CharField(max_length=128)
+    type = models.CharField(max_length=64)
+    timestamp = models.DateTimeField(default=timezone.now, db_index=True)
+    headers = JSONField()
+    blob = FlexibleForeignKey('sentry.FileBlob', null=True)
+
+    # <Legacy fields>
+    storage = models.CharField(max_length=128, null=True)
+    storage_options = JSONField()
+    path = models.TextField(null=True)
+    size = BoundedPositiveIntegerField(null=True)
+    checksum = models.CharField(max_length=40, null=True)
+    # </Legacy fields>
+
+    class Meta:
+        app_label = 'sentry'
+        db_table = 'sentry_file'
+
+    def getfile(self, *args, **kwargs):
+        if not self.blob:
+            lock_key = 'fileblob:convert:{}'.format(self.checksum)
+            with Lock(lock_key):
+                self.update(
+                    blob=FileBlob.objects.get_or_create(
```

Comment:
`get_or_create` also returns an `(obj, created)` tuple which this isn't accounting for.